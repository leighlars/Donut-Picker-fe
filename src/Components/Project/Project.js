import React, { Component } from 'react';
import ReactModal from 'react-modal';
import MiniDonut from  '../MiniDonut/MiniDonut';

class Project extends Component {
  constructor() {
    super();
    this.state = {
      showModal: false,
      currentPalette: ''
    }
  }

  handleOpenModal = (e) => {
    const { id } = e.target
    this.setState({ showModal: true });
    let palette = this.props.palettes.find(palette => palette.id === parseInt(id));
    this.setState({currentPalette: palette})
  }
  
  handleCloseModal = () => {
    this.setState({ showModal: false });
  }

  render() {

  let palettes = this.props.palettes.filter(palette => {
    return palette.project_id === this.props.id
  })

  let displayDonuts = palettes.map((palette, i) => {
    if (palette[`color_${i+1}`][0] === '#') {
      return (<div className="mini-donut-section">
        <h2 className="mini-palette-name">{palette.name}</h2>
        <i id={palette.id} onClick={this.handleOpenModal} className="fas fa-pencil-alt project-icon edit-palette"></i>
        <MiniDonut fill={palette[`color_${i+1}`]} />
        <MiniDonut fill={palette[`color_${i+2}`]} />
        <MiniDonut fill={palette[`color_${i+3}`]} />
        <MiniDonut fill={palette[`color_${i+4}`]} />
        <MiniDonut fill={palette[`color_${i+5}`]} />
        <MiniDonut fill={palette[`color_${i+6}`]} />
      </div>)
    }
  });

  let displayModal = (<div className="mini-donut-section">
        <h2 className="mini-palette-name">Edit {this.state.currentPalette.name}</h2>
        <input type="text" placeholder={this.state.currentPalette.name} />
        <MiniDonut fill={this.state.currentPalette.color_1} />
        <MiniDonut fill={this.state.currentPalette.color_2} />
        <MiniDonut fill={this.state.currentPalette.color_3} />
        <MiniDonut fill={this.state.currentPalette.color_4} />
        <MiniDonut fill={this.state.currentPalette.color_5} />
        <MiniDonut fill={this.state.currentPalette.color_6} />
      </div>)

  return (
    <section>
      <article className="Project">
        <h4 className="Project-name">
        <i className="fas fa-folder-open project-icon" onClick={this.handleOpenModal}></i>
        {this.props.name}</h4>
        <div className="display-donuts">
          {displayDonuts}
        </div>
      </article>
      <ReactModal 
           isOpen={this.state.showModal}
           contentLabel="Palette Modal">
           <div className="modal-display-donuts">
           <input type="text" placeholder="project" maxLength='25' />
          { this.state.showModal && displayModal }
          </div>
          <button onClick={this.handleCloseModal}>Close</button>
        </ReactModal>
    </section>
    )
  }
}


export default Project;
