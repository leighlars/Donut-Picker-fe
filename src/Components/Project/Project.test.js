import React from 'react'
import {shallow} from 'enzyme'
import Project from './Project'

describe('Project', () => {
    let wrapper;
    let mockProps = [{color_1: "#FFF", color_2: "#FFF", color_3: "#FFF"}]
    wrapper = shallow(<Project palettes = {mockProps}/>)
    it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
        
    });
});