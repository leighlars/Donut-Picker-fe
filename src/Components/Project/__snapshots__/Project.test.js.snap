// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project should match snapshot 1`] = `ShallowWrapper {}`;
